src/main/java/org/folio/circulation/resources/renewal/RenewalResource.java

```diff
@@ -142,11 +143,14 @@ private void renew(RoutingContext routingContext) {
       itemRepository, userRepository, loanRepository);
     final var requestQueueRepository = new RequestQueueRepository(requestRepository);
     final LoanPolicyRepository loanPolicyRepository = new LoanPolicyRepository(clients);
+    final OverdueFinePolicyRepository overdueFinePolicyRepository = new OverdueFinePolicyRepository(clients);
```

Comment:
You're creating a second OverdueFinePolicyRepository here (check processFeesFinesForRegularRenew() method). Each repo should preferably have only one instance, so you can create it here and pass it to processFeesFinesForRegularRenew() in the same way as itemRepository is being passed right now.

Reply:
Done but see comments below regarding the alternative (putting the policy in the context rather than passing the repository around)